homeassistant/components/mqtt/light/schema_basic.py

```diff
@@ -500,6 +499,7 @@ def _rgbx_received(
 
         @callback
         @log_messages(self.hass, self.entity_id)
+        @write_state_on_attr_change(self, {"_attr_rgb_color"})
```

Comment:
This is not correct, receiving an rgb / rgbw / rgbww color also sets brightness and color mode depending on the light's configuration.